### PR TITLE
600 PSSS editable number of sites

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- Make Sites Reported an editable field

---

### Screenshots:
Screen Shot 2020-09-11 at 10 21 39 AM

### CODE REVIEW

File: packages/ui-components/src/components/Button.js [hunk 1]
```diff
@@ -173,8 +173,9 @@ export const GreyOutlinedButton = styled(OutlinedButton)`
   padding: 0.5em 1.5em;
 
   &.Mui-disabled {
-    background: ${props => props.theme.palette.grey['100']};
-    border: none;
+    color: ${props => props.theme.palette.text.secondary};
+    background: #f1f1f1;
```
Comment (edmofro) [inline packages/ui-components/src/components/Button.js:177]: f1f1f1 isn't part of the palette?

File: packages/ui-components/src/components/Table/EditableTable.js [hunk 2]
```diff
@@ -55,7 +55,7 @@ const EditableCell = React.memo(({ id, columnKey }) => {
   const key = `${id}-${columnKey}`;
   const value = fields[key];
 
-  if (!value) {
+  if (value === undefined) {
```
Comment (edmofro) [inline packages/ui-components/src/components/Table/EditableTable.js:58]: Oh yeah good catch :+1:

File: packages/psss/src/containers/Tables/CountryReportTable.js [hunk 3]
```diff
@@ -58,36 +87,57 @@ export const CountryReportTableComponent = React.memo(
...
+              <ReportedSites variant="h6">Reported Sites:</ReportedSites>
+              <StyledTextField
+                value={sitesReportedValue}
+                onChange={event => setSitesReportedValue(event.target.value)}
+                name="sites-reported"
+              />
+              <ReportedSites variant="h6"> / Total Sites: 10</ReportedSites>
+            </FormRow>
+          ) : (
+            <Typography variant="h5">7/{sitesReportedValue} Sites Reported</Typography>
```
Comment (edmofro) [inline packages/psss/src/containers/Tables/CountryReportTable.js:111]:
```suggestion
            <Typography variant="h5">{sitesReportedValue}/10 Sites Reported</Typography>
```
Though also the '10' should be coming from the back end at some point

File: packages/psss/src/containers/Tables/CountryReportTable.js [hunk 4]
```diff
@@ -58,36 +87,57 @@ export const CountryReportTableComponent = React.memo(
...
             onClick={handleEdit}
             disabled={tableStatus === TABLE_STATUSES.EDITABLE}
           >
             Edit
           </GreyOutlinedButton>
-        </LayoutRow>
+        </FlexSpaceBetween>
+        {tableStatus === TABLE_STATUSES.EDITABLE && (
+          <Alert severity="error" variant="standard">
+            Updating aggregated data will be the source of truth. All individual Sentinel data will
```
Comment (edmofro) [inline packages/psss/src/containers/Tables/CountryReportTable.js:122]:
```suggestion
            Updating country level data manually: all individual sentinel site data will
```

Comment (edmofro) [reply packages/psss/src/containers/Tables/CountryReportTable.js:122]: Dave suggested this red alert during edit, plus another grey alert that should show while the table is static, if the country level data has been manually edited - is that grey one captured in another card somewhere?

**Review by edmofro — CHANGES_REQUESTED** [review]
Looks good, just some little things

File: packages/psss/src/containers/Tables/CountryReportTable.js (hunk 4 above)
Comment (tcaiger) [reply packages/psss/src/containers/Tables/CountryReportTable.js]: I've updated the wording here to match the design update and added the grey alert ui component. There is still some work to do to finish the behaviour of the table (which data to show, when to show the alert and the reset button) but I think that is dependant on the backend and so I have created another card for it.

File: packages/psss/src/api/FakeApi.js [hunk 5]
```diff
@@ -138,6 +138,7 @@ export class FakeAPI {
         min: 0,
         max: 30,
       }),
+      totalSites: 30,
```
Comment (tcaiger) [inline packages/psss/src/api/FakeApi.js:141]: Added totalSites as a field in the dummy api

File: packages/psss/src/containers/Tables/CountryReportTable.js [hunk 6]
```diff
@@ -26,17 +34,45 @@ const VerifiableBody = props => {
...
 
 const GreyHeader = styled(FakeHeader)`
   border: none;
 `;
 
+const Alert = styled(SmallAlert)`
+  margin-bottom: 1.2rem;
+`;
+
+const GreyAlert = styled(SmallAlert)`
```
Comment (tcaiger) [inline packages/psss/src/containers/Tables/CountryReportTable.js:50]: I have just styled the grey alert as a one off as I think it is too specific for ui-components.

**Comment by tcaiger** [comment]
@edmofro Just making sure that you have seen that this PR is ready for re-review. Cheers

**Comment by edmofro** [comment]
Ah I'd missed this, thanks for the bump

File: packages/psss/src/containers/Tables/CountryReportTable.js [hunk 7]
```diff
@@ -117,11 +127,14 @@ export const CountryReportTableComponent = React.memo(
...
-        {tableStatus === TABLE_STATUSES.EDITABLE && (
+        {tableStatus === TABLE_STATUSES.EDITABLE ? (
           <Alert severity="error" variant="standard">
-            Updating aggregated data will be the source of truth. All individual Sentinel data will
-            be ignored
+            Updating country level data manually: all individual sentinel site data will be ignored
           </Alert>
+        ) : (
+          <GreyAlert severity="info" icon={<InfoIcon fontSize="inherit" />}>
+            Country level data has been manually edited, sentinel data will not be used.
```
Comment (edmofro) [inline packages/psss/src/containers/Tables/CountryReportTable.js:136]: We'll only want to show this under a specific condition, which I guess we'll have to return as a boolean from the back end: has the country level data actually been manually edited

Comment (edmofro) [reply packages/psss/src/containers/Tables/CountryReportTable.js:136]: Can you add that in to the fake api?

**Review by edmofro — APPROVED** [review]
Have one suggestion but you're welcome to make it a separate card if you think this PR has gone on long enough

**Comment by tcaiger** [comment]
> Have one suggestion but you're welcome to make it a separate card if you think this PR has gone on long enough

Cool - yeah I have already created a seperate card for it. I left a comment above about that which must have gotten lost in the long chat but yeah there are a few things to figure out about how that works when the backend is plugged in.

**Pull request merged**